pkg/controller/login/reset_password.go

```diff
@@ -59,6 +60,11 @@ func (c *Controller) HandleSubmitResetPassword() http.Handler {
 			return
 		}
 
+		membership := controller.MembershipFromContext(ctx)
+		if !membership.Can(rbac.UserWrite) {
```

Comment:
Doesn't this method also get called if the user resets their own password? That's why I didn't enforce rbac here.